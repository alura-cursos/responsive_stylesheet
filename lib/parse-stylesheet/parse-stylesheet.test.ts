import parseStyleSheet from "./parse-stylesheet";

describe('parseStyleSheet()', () => {
  describe('when receive default stylesheet rules', () => {
    it('receiving only 1 stylesheet rule', () => {
      const input = {
        backgroundColor: 'red',
      };

      expect(parseStyleSheet(input)).toEqual({
        backgroundColor: 'red',
      });
    });
    it('receiving only 2 stylesheet rules', () => {
      const input = {
        backgroundColor: 'red',
        color: 'white',
      };

      expect(parseStyleSheet(input)).toEqual({
        backgroundColor: 'red',
        color: 'white',
      });
    });
  });
  describe('when receive responsive stylesheet rules', () => {
    it('receiving only 1 stylesheet rule with 1 breakpoint', () => {
      const input = {
        backgroundColor: {
          xs: 'red',
        },
      };

      expect(parseStyleSheet(input)).toMatchSnapshot();
    });
    it('receiving only 1 stylesheet rule with 2 breakpoints', () => {
      const input = {
        backgroundColor: {
          xs: 'red',
          md: 'black',
        },
      };
      expect(parseStyleSheet(input)).toMatchSnapshot();
    });
    it('receiving only a stylesheet rule vertical and horizontal with 2 breakpoints', () => {
      const input = {
        paddingVertical: {
          xs: '10px',
          md: '20px',
        },
        backgroundColor: {
          xs: 'red',
          lg: 'black',
        },
      };

      expect(parseStyleSheet(input)).toMatchSnapshot();
    });
  });
});
