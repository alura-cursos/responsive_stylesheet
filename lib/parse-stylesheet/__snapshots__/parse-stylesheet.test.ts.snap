// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseStyleSheet() when receive responsive stylesheet rules receiving only 1 stylesheet rule with 1 breakpoint 1`] = `
Object {
  "@media (min-width: 0px)": Object {
    "backgroundColor": "red",
  },
}
`;

exports[`parseStyleSheet() when receive responsive stylesheet rules receiving only 1 stylesheet rule with 2 breakpoints 1`] = `
Object {
  "@media (min-width: 0px)": Object {
    "backgroundColor": "red",
  },
  "@media (min-width: 768px)": Object {
    "backgroundColor": "black",
  },
}
`;

exports[`parseStyleSheet() when receive responsive stylesheet rules receiving only a stylesheet rule vertical and horizontal with 2 breakpoints 1`] = `
Object {
  "@media (min-width: 0px)": Object {
    "backgroundColor": "red",
    "padding-bottom": "10px",
    "padding-top": "10px",
  },
  "@media (min-width: 768px)": Object {
    "padding-bottom": "20px",
    "padding-top": "20px",
  },
  "@media (min-width: 992px)": Object {
    "backgroundColor": "black",
  },
}
`;
